infrastructure/smart-contracts/contracts/RefereeCalculations.sol

```diff
@@ -227,4 +234,61 @@ contract RefereeCalculations is Initializable, AccessControlUpgradeable {
 
         return winningKeyCount;
     }
+
+    /**
+     * @notice Retrieves the confirm data for multiple assertions based on their IDs.
+     * @param _assertionIds An array of assertion IDs for which the confirm data is to be retrieved.
+     * @return confirmData An array of `bytes32` values representing the confirm data for each assertion ID provided.
+     *
+     * @dev This function interacts with the `Referee9` contract to retrieve the confirm data for multiple assertions.
+     * It first checks that each assertion ID is valid (i.e., it is less than or equal to the current challenge ID).
+     * Then, it retrieves the `assertionStateRootOrConfirmData` from the corresponding challenge and stores it in the `confirmData` array.
+     *
+     * Note: Ensure that the `refereeAddress` is correctly set before calling this function.
+     */
+    function getConfirmDataMultipleAssertions(
+        uint256[] memory _assertionIds
+    ) public view returns (bytes32[] memory confirmData) {
+        // Create an instance of the Referee9 contract
+        Referee9 referee = Referee9(refereeAddress);
```

Comment:
I don't think we want to add the refereeAddress to the state, we could just pass it to this function.